server/src/test/java/io/spine/server/procman/ProcessManagerRepositoryTest.java

```diff
@@ -648,6 +655,78 @@ void whenCreating() {
+        TestProcessManager pm = new TestProcessManager(id);
+        repository.store(pm);
+
+        // Init filters by the `id_string` column.
+        TargetFilters targetFilters = targetFilters(Project.Column.idString(), id.toString());
+
+        // Check nothing is found as column now should be empty.
+        Iterator<TestProcessManager> found =
+                repository.find(targetFilters, ResponseFormat.getDefaultInstance());
+        assertThat(found.hasNext()).isFalse();
+
+        // Apply the migration.
+        repository.applyMigration(id, new SetTestProcessId());
+
+        // Check the entity is now found by the provided filters.
+        Iterator<TestProcessManager> afterMigration =
+                repository.find(targetFilters, ResponseFormat.getDefaultInstance());
+        assertThat(afterMigration.hasNext()).isTrue();
+
+        // Check the new entity state has all fields updated as expected.
+        TestProcessManager entityWithColumns = afterMigration.next();
+        Project expectedState = pm
+                .state()
+                .toBuilder()
+                .setIdString(pm.getIdString())
+                .build();
+        assertThat(entityWithColumns.state()).isEqualTo(expectedState);
+    }
+
+    @Test
+    @DisplayName("update multiple entities via a custom migration")
+    void performCustomMigrationForMultiple() {
+        // Store three entities to the repository.
+        ProjectId id1 = createId(1);
+        ProjectId id2 = createId(2);
+        ProjectId id3 = createId(3);
+        TestProcessManagerRepository repository = repository();
+        TestProcessManager pm1 = new TestProcessManager(id1);
+        TestProcessManager pm2 = new TestProcessManager(id2);
+        TestProcessManager pm3 = new TestProcessManager(id3);
+        repository.store(pm1);
+        repository.store(pm2);
+        repository.store(pm3);
+
+        // Init filters by the `name` column.
+        TargetFilters filters = targetFilters(Project.Column.name(), SetTestProcessName.NEW_NAME);
+
+        // Check nothing is found as the entity states were not yet updated.
+        Iterator<TestProcessManager> found =
+                repository.find(filters, ResponseFormat.getDefaultInstance());
+        assertThat(found.hasNext()).isFalse();
+
+        // Apply the column update to two of the three entities.
+        repository.applyMigration(ImmutableSet.of(id1, id2), new SetTestProcessName());
+
+        // Check the entities are now found by the provided filters.
+        Iterator<TestProcessManager> foundAfterMigration =
+                repository.find(filters, ResponseFormat.getDefaultInstance());
+
+        ImmutableList<TestProcessManager> results = ImmutableList.copyOf(foundAfterMigration);
+        assertThat(results).hasSize(2);
+        assertThat(results)
+                .comparingElementsUsing(stateWithPropagatedName())
+                .containsExactly(pm1, pm2);
```

Comment:
So the migration does nothing? We're comparing the results to the same PMs we've manually stored.

Reply:
OK, now I see that inside the correspondence we actually manually change the state of the PMs to match the actual one. It looks a bit too complicated...